src/components/views/create/CreateProject.tsx

```diff
@@ -110,19 +109,23 @@ const CreateProject: FC<ICreateProjectProps> = ({ project }) => {
 		categories: storageCategories,
 		impactLocation: storageImpactLocation,
 		image: storageImage,
-		addresses: storageAddresses = {},
+		addresses: storageAddresses = [],
 	} = storageProjectData || {};
 
 	const isDraft = project?.status.name === EProjectStatus.DRAFT;
 	const defaultImpactLocation = impactLocation || '';
-	const activeAddresses = addresses?.filter(a => a.isRecipient) || [];
-
-	const userAddresses = [...new Set(activeAddresses.map(a => a.address!))];
-
-	const addressesObj: { [key: number]: string } = {};
-	activeAddresses.forEach(a => {
-		addressesObj[a.networkId!] = a.address!;
-	});
+	const activeAddresses =
+		addresses
+			?.filter(a => a.isRecipient)
+			.map(a => {
+				const _a = { ...a };
+				delete _a.isRecipient;
+				return _a;
+			}) || [];
+
+	const userUniqueAddresses = [
+		...new Set(activeAddresses.map(a => a.address!)),
+	];
```

Comment:
Please revert changes are not part of solana support

Reply:
it's part of solana support. previously we had chain ids only, now we have chainType too and we can't store addresses in an object like before

Reply:
As we talked, we can store chainType as key for non-evm addresses.